x-pack/plugins/security_solution/public/timelines/components/timeline/query_bar/eql/index.tsx

```diff
@@ -144,7 +147,7 @@ export const EqlQueryBarTimeline = memo(({ timelineId }: { timelineId: string })
 
   useEffect(() => {
     const { index: indexField } = getFields();
-    const newIndexValue = selectedPatterns.sort();
+    const newIndexValue = [...selectedPatterns].sort();
```

Comment:
NIT: doesn't sort already return a new array?

Reply:
no, it mutates the original one